src/dotnet/Telemetry.cs

```diff
@@ -45,6 +70,8 @@ public Telemetry()
                 _commonProperties.Add(OSPlatform, runtimeEnvironment.OperatingSystemPlatform.ToString());
                 _commonProperties.Add(RuntimeId, runtimeEnvironment.GetRuntimeIdentifier());
                 _commonProperties.Add(ProductVersion, Product.Version);
+                _commonProperties.Add(CITestMachine, isCITestMachine.ToString());
```

Comment:
This seems wasteful given the number of times it would occur in the grand scheme.  Can telemetry just be turned off for CI or filtered on the back end (e.g. via IP)?